include/nova_renderer/rhi_types.hpp

```diff
@@ -37,7 +38,7 @@ namespace nova::renderer::rhi {
     };
 
     struct Buffer : Resource {
-        uint64_t size = 0;
+        uint64_t offset = 0u, size = 0u; // added offset, because Vulkan API may use different address in allocated memory
```

Comment:
The offset for Vulkan memory allocations (or any other allocations) is stored in a `DeviceMemoryAllocation` struct

If it wasn't, Vulkan-specific data for buffers should be stored in `VulkanBuffer`, which already has some Vulkan-specific data such as the `VkBuffer` itself